### DISCUSSION TITLE
Tycho build broken in GitHub workflow

### DISCUSSION THREAD

**Discussion by guw** [eclipse-tycho/tycho#3834] [Q&A]
I have a Tycho build that stopped working in GitHub. I don't get the failure locally so I suspect an environment issue. If anyone has any pointers I'd appreciate it.

The error message is:
```
Caused by: org.apache.maven.plugin.MojoFailureException: 
  An unexpected error occurred while launching the test runtime (process returned error code 13). 
  Command-line used to launch the sub-process was 

/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.11-9/arm64/Contents/Home/bin/java -Dosgi.noShutdown=false -Dosgi.os=macosx -Dosgi.ws=cocoa -Dosgi.arch=aarch64 -XstartOnFirstThread -Xmx1024m -Dorg.eclipse.swtbot.search.timeout=30000 -Dorg.eclipse.swtbot.screenshots.dir=target/screenshots -Dosgi.clean=true -Dosgi.console=1234 -jar /Users/runner/.m2/repository/p2/osgi/bundle/org.eclipse.equinox.launcher/1.6.600.v20231106-1826/org.eclipse.equinox.launcher-1.6.600.v20231106-1826.jar -data /Users/runner/work/bazel-eclipse/bazel-eclipse/tests/com.salesforce.bazel.sdk.tests/target/work/data -install /Users/runner/work/bazel-eclipse/bazel-eclipse/tests/com.salesforce.bazel.sdk.tests/target/work -configuration /Users/runner/work/bazel-eclipse/bazel-eclipse/tests/com.salesforce.bazel.sdk.tests/target/work/configuration -application org.eclipse.tycho.surefire.osgibooter.uitest -testproperties /Users/runner/work/bazel-eclipse/bazel-eclipse/tests/com.salesforce.bazel.sdk.tests/target/surefire.properties in working directory /Users/runner/work/bazel-eclipse/bazel-eclipse/tests/com.salesforce.bazel.sdk.tests
```

I tried updating to latest Tycho (now 4.0.7) and Maven (now 3.9.6) without success.

I recall the `returned error code 13` implies that the Equinox launcher is requesting a restart. But I wonder why.

**Comment by laeubi** [comment]
> I recall the returned error code 13 implies that the Equinox launcher is requesting a restart. But I wonder why.

The Equinox Exit codes are

- `EXIT_OK` (0)
- `EXIT_RESTART` (23)
- `EXIT_RELAUNCH` (24)

So `13` seems to mean something like [invalid VM configured](https://stackoverflow.com/questions/4945178/cannot-run-eclipse-jvm-terminated-exit-code-13), so you should watch out for additional warnings in the log, maybe it is because Github has [switched to arm based mac nodes](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) recently with the `macos-latest` so you maybe want to try the `macos-13` in the meanwhile.

**Reply by guw** [reply]
That's a good hint. Thanks @laeubi I missed that announcement and so some more debugging.
